crates/scarb-api/src/artifacts.rs

```diff
@@ -92,3 +91,107 @@ fn compile_artifact_at_path(path: &Utf8Path) -> Result<StarknetContractArtifacts
 
     Ok(StarknetContractArtifacts { sierra, casm })
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::ScarbCommand;
+    use camino::Utf8PathBuf;
+    use deserialized::{StarknetArtifacts, StarknetContract, StarknetContractArtifactPaths};
+
+    #[test]
+    fn test_unique_artifacts() {
+        // Mock StarknetArtifactsRepresentation
+        let mock_base_artifacts = HashMap::from([(
+            "contract1".to_string(),
+            (
+                StarknetContractArtifacts {
+                    sierra: "sierra1".to_string(),
+                    casm: "casm1".to_string(),
+                },
+                Utf8PathBuf::from("path1"),
+            ),
+        )]);
+
+        let mock_representation_1 = StarknetArtifactsRepresentation {
+            base_path: Utf8PathBuf::from("mock/path1"),
+            artifacts: StarknetArtifacts {
+                version: 1,
+                contracts: vec![StarknetContract {
+                    id: "1".to_string(),
+                    package_name: "package1".to_string(),
+                    contract_name: "contract1".to_string(),
+                    artifacts: StarknetContractArtifactPaths {
+                        sierra: Utf8PathBuf::from("mock/path1/contract1.sierra"),
+                    },
+                }],
+            },
+        };
+
+        let mock_representation_2 = StarknetArtifactsRepresentation {
+            base_path: Utf8PathBuf::from("mock/path2"),
+            artifacts: StarknetArtifacts {
+                version: 1,
+                contracts: vec![StarknetContract {
+                    id: "2".to_string(),
+                    package_name: "package2".to_string(),
+                    contract_name: "contract2".to_string(),
+                    artifacts: StarknetContractArtifactPaths {
+                        sierra: Utf8PathBuf::from("mock/path2/contract2.sierra"),
+                    },
+                }],
+            },
+        };
+
+        let representations = vec![mock_representation_1, mock_representation_2];
+
+        let result = unique_artifacts(representations, &mock_base_artifacts);
+
+        assert_eq!(result.len(), 1);
+        assert_eq!(result[0].0, "contract2");
+    }
+
+    #[test]
+    #[cfg_attr(not(feature = "scarb_2_8_3"), ignore)]
+    fn test_load_contracts_artifacts() {
+        let temp = crate::tests::setup_package("basic_package");
+
+        ScarbCommand::new_with_stdio()
+            .current_dir(temp.path())
+            .arg("build")
+            .arg("--test")
+            .run()
+            .unwrap();
+
+        // Define path to the generated artifacts
+        let base_artifacts_path = temp.to_path_buf().join("target").join("dev");
+
+        // Get the base artifact
+        let base_file = Utf8PathBuf::from_path_buf(
+            base_artifacts_path.join("basic_package_integrationtest.test.starknet_artifacts.json"),
+        )
+        .unwrap();
```

Comment:
This is missing the logic that creates the test file in `basic_package` that was added at some point.

Reply:
Done @cptartur